kite-data/kite-data-core/src/test/java/org/kitesdk/data/spi/filesystem/TestFileSystemDataset.java

```diff
@@ -609,6 +612,111 @@ public void testDeleteAllWithoutPartitions() {
     checkReaderBehavior(ds.newReader(), 0, (RecordValidator<Record>) null);
   }
 
+  @Test
+  public void signalReadyOnUnboundedDataset() {
+    final FileSystemDataset<Record> ds = new FileSystemDataset.Builder<Record>()
+        .namespace("ns")
+        .name("users")
+        .configuration(getConfiguration())
+        .descriptor(
+            new DatasetDescriptor.Builder().schema(USER_SCHEMA).format(format)
+                .location(testDirectory).build())
+        .type(Record.class)
+        .uri(URIBuilder.build(URI.create("repo:" + testDirectory.toUri()), "ns", "name"))
+        .build();
+    Assert.assertFalse("Unbounded dataset has not been signaled", ds.isReady());
+    ds.signalReady();
+    Assert.assertTrue("Unbounded dataset has been signaled and should be ready", ds.isReady());
+  }
+
+  private static final long NOT_READY_TIMESTAMP = -1L;
+
+  @Test
+  public void testReadySignal() throws InterruptedException {
+    final FileSystemDataset<Record> ds = new FileSystemDataset.Builder<Record>()
+        .namespace("ns")
+        .name("users")
+        .configuration(getConfiguration())
+        .descriptor(
+            new DatasetDescriptor.Builder().schema(USER_SCHEMA).format(format)
+                .location(testDirectory).build())
+        .type(Record.class)
+        .uri(URIBuilder.build(URI.create("repo:" + testDirectory.toUri()), "ns", "name"))
+        .build();
+
+    Constraints viewConstraints = new Constraints(USER_SCHEMA).
+        with("username", "bob");
+
+    Assert.assertEquals("Constraint should not be ready before being signaled",
+        NOT_READY_TIMESTAMP, ds.getReadyTimestamp(viewConstraints));
+    long currentTime = System.currentTimeMillis();
+    // wait some time to ensure we can distinguish the current time vs the signal time
+    Thread.sleep(1000);
+    ds.signalReady(viewConstraints);
+    long signaledTime = ds.getReadyTimestamp(viewConstraints);
+    Assert.assertTrue("Constraint should be ready now", signaledTime != NOT_READY_TIMESTAMP);
+    Assert.assertTrue("Signal should be after or at time as of signaling",
+        signaledTime > currentTime);
+  }
+
+  @Test
+  public void testReadySignalForSpecificConstraints() {
+    final FileSystemDataset<Record> ds = new FileSystemDataset.Builder<Record>()
+        .namespace("ns")
+        .name("users")
+        .configuration(getConfiguration())
+        .descriptor(
+            new DatasetDescriptor.Builder().schema(USER_SCHEMA).format(format)
+                .location(testDirectory).build())
+        .type(Record.class)
+        .uri(URIBuilder.build(URI.create("repo:" + testDirectory.toUri()), "ns", "name"))
+        .build();
+
+    Constraints viewConstraints = new Constraints(USER_SCHEMA).
+        with("username", "bob");
+    Constraints unsignaledConstraints = new Constraints(USER_SCHEMA).
+        with("username", "joe");
+
+    ds.signalReady(viewConstraints);
```

Comment:
Updating to passing the signal manager into the view on construction removed the need for the Dataset.signalReady(constraints) methods and removed the need for some of the tests at this level, the ones that remained made much more sense to use the exposed APIs for the tests. The change can be seen here: 820d9c8f